master/buildbot/test/unit/test_status_builder.py

```diff
@@ -0,0 +1,67 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+"""
+Tests for buildbot.status.builder module.
+"""
+from buildbot.status import builder
+from buildbot.test.fake import fakemaster
+from twisted.trial import unittest
+
+
+class TestBuilderStatus(unittest.TestCase):
+    """
+    Unit tests for BuilderStatus.
+    """
+
+    def makeBuilderStatus(self):
+        """
+        Return a new BuilderStatus.
+        """
+
+        return builder.BuilderStatus(
+            buildername='testing-builder',
+            tags=None,
+            master=fakemaster.make_master(),
+            description=None)
+
+    def test_matchesAnyTag_no_tags(self):
+        """
+        Return False when builder has no tags.
+        """
+        sut = self.makeBuilderStatus()
+
+        self.assertFalse(sut.matchesAnyTag(set()))
+        self.assertFalse(sut.matchesAnyTag(set(('any-tag', 'tag'))))
+
+    def test_matchesAnyTag_no_match(self):
+        """
+        Return False when requested tags don't match.
+        """
+        sut = self.makeBuilderStatus()
+        sut.tags = {'one'}
```

Comment:
There is already one

Le mar. 10 mars 2015 15:20, Adi Roiban <notifications@github.com> a écrit :

> In master/buildbot/test/unit/test_status_builder.py
> <https://github.com/buildbot/buildbot/pull/1576#discussion_r26125132>:
>
> > +
> > +    def test_matchesAnyTag_no_tags(self):
> > +        """
> > +        Return False when builder has no tags.
> > +        """
> > +        sut = self.makeBuilderStatus()
> > +
> > +        self.assertFalse(sut.matchesAnyTag(set()))
> > +        self.assertFalse(sut.matchesAnyTag(set(('any-tag', 'tag'))))
> > +
> > +    def test_matchesAnyTag_no_match(self):
> > +        """
> > +        Return False when requested tags don't match.
> > +        """
> > +        sut = self.makeBuilderStatus()
> > +        sut.tags = {'one'}
>
> true... fixed. Maybe it would be nice to have a tox.ini file or some
> script to simplify testing on different python version.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/buildbot/buildbot/pull/1576/files#r26125132>.
>